### DISCUSSION TITLE
ImGui's example file not working on windows

### DISCUSSION THREAD

**Discussion by ryback08** [cavariux/nimgl#62] [Q&A] [answered]
Hello nim world.

I am trying to run the example file to try imgui.
I compile the example file "timgui" with nim (windows 10, nim 1.4.2, install nimgl with nimble)
I put cimgui.dll ( download here: https://github.com/cimgui/cimgui/releases ) in the same folder.

I see a nice window with imgui but the GUI is not responding. Any reaction when I click on the imgui menu.
Moreover, I don't see any framerate.

have you seen the same behavior.
I don't thing it is a bug, i think a missing something.

Thanks to the community.

**Comment by lmariscal** [comment, marked as answer]
Hi,

I would recommend you to build the official fork of cimgui https://github.com/nimgl/cimgui
Remember to compile using MSVC even if you are running Nim with GCC. Check that the architectures are correct for both Nim and the DLL. To see if the issue is with the library you can try using the C++ backend, since that would compile both cimgui and imgui.

Please let me know more information (any stacktrace? nvidia/amd/intel? drivers?) after doing the previous recommendations, so I can better help you.

**Comment by ryback08** [comment]
Hi Imariscal, better 2021 and thanks for your help.

After a lot of research, I managed to recompile cimgui to DLL and it works !

 I globaly did this
- I personnaly use Visual Studio Code. Install  "C / C ++" and "CMake Addon"
- install “build tool for visual studio” (https://visualstudio.microsoft.com/fr/downloads/)

- Clone with GIT cimgui on your hard drive.
- In VScode, do "open directory" , configure Cmake and Build.

I got help from the following tutorials:
https://www.youtube.com/watch?v=wP4cwAtU-g8
https://code.visualstudio.com/docs/cpp/cmake-linux
